src/Controllers/LoginSessionController.php

```diff
@@ -29,42 +33,48 @@ class LoginSessionController extends Controller
      */
     public function remove(HTTPRequest $request): HTTPResponse
     {
-        return $this->removeLoginSession($request);
-    }
-
-    private function removeLoginSession(HTTPRequest $request): HTTPResponse
-    {
-        $failureMessage = _t(__CLASS__ . '.REMOVE_FAILURE', 'Something went wrong.');
-        try {
-            // Ensure CSRF protection
-            if (!SecurityToken::inst()->checkRequest($request)) {
-                return $this->jsonResponse([
+        if (empty(Security::getCurrentUser())) {
+            return $this->jsonResponse(
+                [
                     'success' => false,
-                    'message' => $failureMessage
-                ]);
-            }
+                    'message' => _t(
+                        __CLASS__ . '.YOUR_SESSION_HAS_EXPIRED',
+                        'Your session has expired'
+                    )
+                ],
+                401
+            );
+        }
 
-            $id = $request->param('ID');
-            $loginSession = LoginSession::get()->byID($id);
-            if (!$loginSession) {
-                return $this->jsonResponse([
+        // Validate CSRF token and ID parameter
+        $id = $request->param('ID');
+        if (!SecurityToken::inst()->checkRequest($request) || empty($id) || !is_numeric($id)) {
+            return $this->jsonResponse(
+                [
                     'success' => false,
-                    'message' => $failureMessage
-                ]);
-            }
+                    'message' => _t(
+                        __CLASS__ . '.INVALID_REQUEST',
+                        'Invalid request'
```

Comment:
I'm adding an explicit check to make sure ID makes sense. In practice there's no way a user should be able to reach this  during normal operation.